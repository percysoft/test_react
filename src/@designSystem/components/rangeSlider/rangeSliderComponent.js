import React from 'react';
import Slider from '@material-ui/core/Slider';
import { withStyles } from '@material-ui/core/styles';
import * as PropTypes from "prop-types";
import { TextStyle } from '../../styles';

const RangeSlider = withStyles({
  root: {
    color: '#8230E0',
    height: 8,
  },
  thumb: {
    height: 24,
    width: 24,
    backgroundColor: '#fff',
    border: '1px solid #D8D8D8',
    marginTop: -8,
    marginLeft: -12,
    '&:focus, &:hover, &$active': {
      boxShadow: 'inherit',
    },
  },
  active: {},
  valueLabel: {
    left: 'calc(-50% + 4px)',
  },
  track: {
    height: 8,
    borderRadius: 4,
  },
  rail: {
    height: 8,
    borderRadius: 4,
  },
})(Slider);

export const RangeSliderComponent = (props) => {

  const { minQuotas , maxQuotas, setValuoQuota, valueQuota,disabled} = props;

  const onChange =  (e, value) => {
    setValuoQuota(value);
  }
  return(
    <div>
      <RangeSlider 
        key='slider-range'
        valueLabelDisplay="auto"
        defaultValue={valueQuota}
        value={valueQuota}
        min={minQuotas}
        max ={maxQuotas}
        onChange={onChange}
        disabled={disabled}

      />
      <div style={{display: 'flex', justifyContent:'space-between'}}>
        <TextStyle bold="800" size="25" color="#8230E0"> {`${valueQuota} cuotas`}</TextStyle>
        <TextStyle bold="800" size="25"> {`MÃ¡ximo: ${maxQuotas}`}</TextStyle>
      </div>
    </div>
  )
}

RangeSliderComponent.propTypes = {
  minQuotas: PropTypes.number,
  maxQuotas: PropTypes.number,
};

RangeSliderComponent.defaultProps = {
  minQuotas: 1,
  maxQuotas: 50,
  valueQuota: 1,
};
