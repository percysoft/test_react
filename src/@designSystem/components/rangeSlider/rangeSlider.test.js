import React from 'react';
import '@testing-library/jest-dom';
import { RangeSliderComponent } from './rangeSliderComponent';
import { shallow } from 'enzyme';

describe('Test Range Slider Component Component', () => {
  test('Show value quotas', () => {
    let valueQuota= 10000
    const  wrapper = shallow (<RangeSliderComponent valueQuota={valueQuota} />)
    expect(wrapper).toMatchSnapshot();
  })
})