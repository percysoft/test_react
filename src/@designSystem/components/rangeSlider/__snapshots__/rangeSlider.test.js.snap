// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Range Slider Component Component Show value quotas 1`] = `
<div>
  <WithStyles(WithStyles(ForwardRef(Slider)))
    defaultValue={10000}
    key="slider-range"
    max={50}
    min={1}
    onChange={[Function]}
    value={10000}
    valueLabelDisplay="auto"
  />
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <styled.div
      bold="800"
      color="#8230E0"
      size="25"
    >
       
      10000 cuotas
    </styled.div>
    <styled.div
      bold="800"
      size="25"
    >
       
      MÃ¡ximo: 50
    </styled.div>
  </div>
</div>
`;
